### ISSUE TITLE
blacksheep.client add any request method support

### ISSUE THREAD

**Issue by ChenyangGao** [Neoteroi/BlackSheep#517]
##### _Note: consider using [Discussions](https://github.com/Neoteroi/BlackSheep/discussions) to open a conversation about new features…_

**🚀 Feature Request**

At present, `blacksheep.client` only supports some common request methods. I hope to accept any method, such as PROPFIND (in webdav)

**Comment by RobertoPrevato** [comment]
Hi @ChenyangGao 
Thank you for opening this issue, and apologies for taking so long to reply.
Good idea, the code API can be modified to support that kind of scenario. For now, you can achieve what you desire easily by defining your own `ClientSession` this way:

```python
from blacksheep.client import ClientSession
from blacksheep.common.types import HeadersType, ParamsType, URLType, normalize_headers
from blacksheep.messages import Request, Response


class MyClientSession(ClientSession):
    async def send_with_details(
        self,
        method: str,
        url: URLType,
        headers: HeadersType | None = None,
        params: ParamsType | None = None,
    ) -> Response:
        return await self.send(
            Request(method, self.get_url(url, params), normalize_headers(headers))
        )
```

When coding I intentionally keep most members public to help people subclassing when they desire to change the behavior of the class. Please don't be afraid of subclassing and changin the behavior when you need doing so (don't listen to those who cry _composition is better than inheritance_ - there is nothing wrong in subclassing and `super()`).

**Comment by ChenyangGao** [comment]
OK, and thank you very much for your answers

**Comment by ChenyangGao** [comment]
@RobertoPrevato 

https://github.com/Neoteroi/BlackSheep/blob/main/blacksheep/scribe.pyx#L47-L57

There is an `HTTP_METHODS` dictionary that restricts all possible values, but no PROPFIND

Perhaps it should be deleted, and each `HTTP_METHODS[request.method]`  is replaced with `get_method(request)`

`get_method` defines below

```python
cpdef bytes get_method(Request request):
    return request.method.encode("ascii")
```

**Comment by RobertoPrevato** [comment]
Yes you are correct, the change you propose is correct. When I initially replied to this issue I didn't recall about that detail, sorry.

**Comment by RobertoPrevato** [comment]
@ChenyangGao 
I fixed the problem in a branch. For now, as a temporary workaround before I publish a new version to PyPi, you could alter the `HTTP_METHODS`  dictionary this way:

```python
from blacksheep.messages import Request
from blacksheep.scribe import write_small_request, HTTP_METHODS  # type: ignore


HTTP_METHODS.update({"PROPFIND": b"PROPFIND"})


request = Request("PROPFIND", b"/", [])
print(write_small_request(request))
```

**Comment by ChenyangGao** [comment]
Amazing, so that I can write a webdav server with BlackSheep.
